schemaregistry/schemaregistry.md

```diff
@@ -143,33 +146,102 @@ Conceptually, a schema is a description of a data structure. Since data
 structures evolve over time, the schema describing them will also evolve over
 time. Therefore, a schema often has multiple versions.
 
-For simple scenarios, the API allows for version management to be automatic and
-transparent. Whenever a schema is updated, a new version number is assigned and
-prior schema versions are retained. This specification does not mandate a
-retention policy, but implementations MAY retire and remove outdated schema
-versions.
-
-The latest available schema is always the default version that is retrieved when
-the URL to the schema is given without the version specified.
+In this specification, the ["schema version"](#223-schema-version-attributes)
+refers to an individual schema document. The ["schema"](#224-schema-attributes)
+is a management bracket for those documents that helps enforcing consistency
+across versions, including compatibility policies.
 
 A newer schema version might introduce breaking changes or it might only
 introduce careful changes that preserve compatibility. These strategies are not
 subject of this specification, but the API provides a conflict handling
 mechanism that allows an implementation to reject updates that do not comply
 with a compatibility policy, if one has been implemented.
 
-### 2.2.1. Multi-format schema groups
+For simple scenarios, the API allows for version management to be automatic and
+transparent. Whenever a schema is updated, a new version number is assigned and
+prior schema versions are retained. This specification does not mandate a
+retention policy, but implementations MAY retire and remove outdated schema
+versions.
+
+The latest available schema is always the default version that is retrieved when
+the URI to the schema is used without the version having been specified.
+
+### 2.2.1 Schema authority
```

Comment:
WDYT about removing the Schema Version URI concept and make the schema authority a parameter in `getSchemaVersion` api operation, or just create an api operation where authority is a explicit parameter...

Here two examples showing the CloudEvent `dataschema` attribute and how I suggest the Schema Authority could be implemented.

##### Example 1 (explicit authority)
`dataschema: corp.example.com/mygroup/schemas/myschema/versions/2`

`authority: corp.example.com`
`schema-group: mygroup`
`schema-id: myschema`
`schema-version: 2`

##### Example 2 (implicit authority)
`dataschema: /mygroup/schemas/myschema/versions/2`

`authority: `(not specified - schema registry should return it's default implicit authority)
`schema-group: mygroup`
`schema-id: myschema`
`schema-version: 2`

##### Option 1
both cases will use `getSchemaVersion` api operation 
`getSchemaVersion` will accept a new header in which authority can be specified

##### Option 2
add a new api operation with request path like:
`authorities/{authority}/schemagroups/{group-id}/schemas/{schema-id}/versions/{version-number}`
for cases when no authority is specified in the dataschema a `default` authority could be used

Reply:
The Schema Version URI exists such that we can have a unique identifier for a schema that works for lookup of the schema independent of which registry contains it: It is a unique identifier. 

"dataschema" must have an absolute URI to reference, because events get forwarded. If you publish into an event broker and that forwards into Kafka and then you have another event broker that also forwards into Kafka, the Kafka consumer has no idea what a relative URI means for "dataschema"

The way you would set up that topology is for there to be a registry that belongs with the Kafka broker and that has replication relationships with those two upstream brokers' registries. Since the upstream event brokers otherwise have no relationship except feeding into the same Kafka brokers, we need to disambuigate the authorities and that needs to be reflected in "dataschema"